src/data_factory_testing_framework/_test_framework.py

```diff
@@ -146,10 +157,17 @@ def evaluate_activities(self, activities: List[Activity], state: PipelineRunStat
                     activities_iterator = []
                     if isinstance(activity, ExecutePipelineActivity) and self.should_evaluate_child_pipelines:
                         execute_pipeline_activity: ExecutePipelineActivity = activity
-                        pipeline = self.get_pipeline_by_id(
-                            # Note: in the future they will probably rename this to referenceId for Fabric
-                            execute_pipeline_activity.type_properties["pipeline"]["referenceName"],
-                        )
+                        pipeline: Pipeline = None
+                        if self.framework_type == TestFrameworkType.Fabric:
+                            pipeline = self.get_pipeline_by_id(
+                                # Note: in the future they will probably rename this to referenceId for Fabric
+                                execute_pipeline_activity.type_properties["pipeline"]["referenceName"],
+                            )
+                        else:
+                            pipeline = self.get_pipeline_by_name(
+                                # Note: in the future they will probably rename this to referenceId for Fabric
```

Comment:
You can remove this comment here as it's not applicable for the other TestFrameworkTypes

Reply:
Done.